dockerHashing/src/main/scala/cromwell/docker/registryv2/DockerRegistryV2Abstract.scala

```diff
@@ -0,0 +1,285 @@
+package cromwell.docker.registryv2
+
+import cats.data.NonEmptyList
+import cats.effect.IO
+import cats.syntax.either._
+import common.validation.Validation._
+import cromwell.docker.DockerInfoActor._
+import cromwell.docker.registryv2.DockerRegistryV2Abstract._
+import cromwell.docker.{DockerHashResult, DockerImageIdentifier, DockerRegistry, DockerRegistryConfig}
+import io.circe.Decoder
+import io.circe.generic.auto._
+import org.http4s.Uri.{Authority, Scheme}
+import org.http4s._
+import org.http4s.circe.{CirceInstances, _}
+import org.http4s.client.Client
+import org.http4s.client.dsl.io._
+import org.http4s.headers.{Accept, _}
+import org.http4s.util.CaseInsensitiveString
+
+object DockerRegistryV2Abstract {
+  implicit class EnhancedParseResult[A](val parseResult: ParseResult[A]) extends AnyVal {
+    def unsafe(context: String): A =
+      parseResult
+        .leftMap(_.message)
+        .leftMap(NonEmptyList.one)
+        .unsafe(context)
+  }
+
+  val DigestHeaderName = CaseInsensitiveString("Docker-Content-Digest")
+  val ManifestV2MediaType = "application/vnd.docker.distribution.manifest.v2+json"
+  val ManifestListV2MediaType = "application/vnd.docker.distribution.manifest.list.v2+json"
+
+  // If one of those fails it means someone changed one of the strings above to an invalid one.
+  val ManifestV2MediaRange = MediaRange.parse(ManifestV2MediaType)
+    .unsafe("Cannot parse invalid manifest v2 content type. Please report this error.")
+  val ManifestListV2MediaRange = MediaRange.parse(ManifestListV2MediaType)
+    .unsafe("Cannot parse invalid manifest list v2 content type. Please report this error.")
+  val AcceptManifestV2Header = Accept.parse(ManifestV2MediaType)
+    .unsafe("Cannot parse invalid manifest v2 Accept header. Please report this error.")
```

Comment:
ToL: I don't know this particular API yet. Does it also log the actual header that failed to parse? Or does the user just get the string above with no other context? I also don't know the headers and if they contain sensitive info that _should not_ be logged.

ToL assuming this code was existing and just moved.